nshlib/nsh_command.c

```diff
@@ -186,7 +186,12 @@ static const struct cmdmap_s g_cmdmap[] =
 #endif
 
 #ifndef CONFIG_NSH_DISABLE_DD
-  CMD_MAP("dd",       cmd_dd,       3, 7,
+  CMD_MAP("dd",       cmd_dd,
+#  ifdef CONFIG_NSH_PIPELINE
```

Comment:
it's better to decouple dd from/to stdin/stdout from CONFIG_NSH_PIPELINE since it doesn't have the direct dependence

Reply:
> it's better to decouple dd from/to stdin/stdout from CONFIG_NSH_PIPELINE since it doesn't have the direct dependence

Get, indeed, thank you!
Selftest updated:
```bash
# 1. Input from stdin and output to stdout
#   Keyboard input: 12345AAAAABBBBB
nsh> dd bs=5
1234512345AAAAAAAAAABBBBBBBBBB

# 2. Input from file and output to stdout
nsh> dd if=/etc/init.d/rc.sysinit
mkrd -m 2 -s 512 1024
mkfatfs /dev/ram2
mount -t vfat /dev/ram2 "/tmp"

# 3. Input from stdin and output to file
#    Keyboard input: QWERT
dd of=/data/dd_stdout bs=5
#    Then, cat the output file in host (based on HostFS):
$ cat ../nuttx/dd_stdout
QWERT
```